src/main/java/com/revature/controllers/AddressController.java

```diff
@@ -0,0 +1,54 @@
+package com.revature.controllers;
+
+import java.util.List;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.revature.models.Address;
+import com.revature.models.AddressDTO;
+import com.revature.services.AddressService;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@RestController
+@RequestMapping("/address")
+@CrossOrigin
+@Api(tags = {"Address"})
+public class AddressController {
+	private AddressService addressService;
+
+	@Autowired
+	public void setAddressService(AddressService addressService) {
+		this.addressService = addressService;
+	}
+	
+	@ApiOperation(value = "Return all addresses", tags = {"Address"})
+	@GetMapping
+	public ResponseEntity<List<AddressDTO>> getAddresses(){
+		List<AddressDTO> dtos = addressService.findAllDTO();
+		if(dtos.isEmpty())return ResponseEntity.notFound().build();
+		return ResponseEntity.ok(dtos);
+	}
+	
+	@ApiOperation(value = "Add a new address", tags = {"Address"})
+	@PostMapping
+	public ResponseEntity<AddressDTO> addAddress(@Valid @RequestBody AddressDTO addressDTO){
+		Address address = addressService.addAddress(new Address(addressDTO));
```

Comment:
We shouldn't save the entity before checking if it was good to do so. We should instead check if an Address of that id already exists, and if it does, we should send back a bad request.

Reply:
I added a check before adding, but still want a last fail point after. Do we want a different status code for the last fail point check?

Reply:
Hmm. I suppose it might still be possible for address to be null, so I suppose that check is okay. I think just a 400 Bad Request is more apt though.